entities/service.go

```diff
@@ -134,3 +131,43 @@ func IncreaseServiceID() {
 	defer svcID.mu.Unlock()
 	svcID.ID++
 }
+
+// CreateServiceFromTemplate creates k8s service based on template
+func CreateServiceFromTemplate(cs *k8sKubernetes.Clientset, t ServiceTemplate) (string, kubernetes.ServiceBase, string, error) {
+	IncreaseServiceID()
+
+	servicePorts := make([]v1.ServicePort, len(t.ProtocolPorts))
+	for _, sp := range t.ProtocolPorts {
+		servicePorts = append(servicePorts, v1.ServicePort{
+			Name:     fmt.Sprintf("service-port-%s-%v", strings.ToLower(string(sp.Protocol)), sp.Port),
+			Protocol: sp.Protocol,
+			Port:     sp.Port,
+		})
+	}
+
+	s := &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-%d", t.Name, svcID.ID),
+			Namespace: t.Namespace,
+		},
+		Spec: v1.ServiceSpec{
+			Selector: t.Selector,
+			Ports:    servicePorts,
+		},
+	}
+	if t.SessionAffinity {
+		s.Spec.SessionAffinity = "ClientIP"
+	}
+
+	var service kubernetes.ServiceBase = kubernetes.NewService(cs, s)
```

Comment:
since this is an entity I Would avoid mixing kubernetes logic on this layer. The idea is a `model` object where we can fill the data structure and treat transformation, k8s calls should go one level up (manager/controllers).

Reply:
moved into manager.